### PR TITLE
Few touches on the Plugin Guid

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Related to #6161

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md [hunk 1]
```diff
@@ -13,14 +13,18 @@ To provide type safety and IntelliSense for your plugin users, please refer to t
 
 ## Plugin Dependencies
 
-At the minimum, your plugin should depend on the `4.x` version of `web3`. This will allow your plugin class to extend the provided `Web3PluginBase` abstract class. However, `web3` shouldn't be listed as a regular dependency, instead it should be listed in your plugin's `package.json` as a [peer dependency](https://nodejs.org/en/blog/npm/peer-dependencies/):
+At the minimum, your plugin should depend `web3` package version `4.0.2`. This will allow your plugin class to extend the provided `Web3PluginBase` abstract class. However, `web3` shouldn't be listed as a regular dependency, instead it should be listed in your plugin's `package.json` as a [peer dependency](https://nodejs.org/en/blog/npm/peer-dependencies/).
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_authors.md]:
```suggestion
At the minimum, your plugin should depend on `web3` package version `4.0.2`. This will allow your plugin class to extend the provided `Web3PluginBase` abstract class. However, `web3` shouldn't be listed as a regular dependency, instead it should be listed in your plugin's `package.json` as a [peer dependency](https://nodejs.org/en/blog/npm/peer-dependencies/).
```

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md [hunk 2]
```diff
@@ -13,14 +13,18 @@ To provide type safety and IntelliSense for your plugin users, please refer to t
...
+
+:::important
+If the version `web3@4.0.2`, was not available yet. You can use the version `web3@4.0.2-dev.af57eae.0`.
+:::
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_authors.md:20]:
```suggestion
```

When this doc is officially release, version `4.0.2` (or maybe even `4.1.0`) will be the latest version available

**Comment by Spacesai1or** [comment]
I still think [this](https://github.com/web3/web3.js/pull/6165#discussion_r1223553869) change should be reverted

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md (hunk 2 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_plugin_guide/plugin_authors.md:20]: Yes, but after being merged with 4.x branch the documentation will be updated and people will read it before the release of 4.0.2 or 4.1.0, right?

**Comment by Muhammad-Altabba** [comment]
> I still think [this](https://github.com/web3/web3.js/pull/6165#discussion_r1223553869) change should be reverted

Thanks for your efforts in keeping and improving the quality. I will revert back to the old text for that paragraph if someone else also has the same thoughts as yours. So, it is like voting :smile: And I hope this is good.

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md (hunk 2 above)
Comment (jdevcs) [reply docs/docs/guides/web3_plugin_guide/plugin_authors.md:20]: https://github.com/web3/web3.js/issues/6191 
I think lets not refer specific dev versions where possible in future, as these are unstable and only for dev testing.

**Review by jdevcs — APPROVED** [review]
lgtm, except some earlier comments.

**Comment by luu-alex** [comment]
In this [discussion](https://github.com/web3/web3.js/pull/6165#discussion_r1223553869) 
For this I think its more clear to me in the original comment 
> In simpler terms, you will be making TypeScript aware that you are modifying the interface of `Web3Context`, and any class that extends it to include the interface of your plugin (i.e. your plugin's added methods, properties, etc.)

rather than the new one

> In simpler terms, this is to modify the `Web3Context` class, and any inheriting classes, to make your plugin's functionality accessible from within.

**Comment by Muhammad-Altabba** [comment]
> In this [discussion](https://github.com/web3/web3.js/pull/6165#discussion_r1223553869) For this I think its more clear to me in the original comment
> 
> > In simpler terms, you will be making TypeScript aware that you are modifying the interface of `Web3Context`, and any class that extends it to include the interface of your plugin (i.e. your plugin's added methods, properties, etc.)
> 
> rather than the new one
> 
> > In simpler terms, this is to modify the `Web3Context` class, and any inheriting classes, to make your plugin's functionality accessible from within.

@luu-alex and @spacesailor24 
I restored the last paragraph. And just added a sentence and 2 phrases :smile: .I hope how it is better. Kindly, recheck.

**Review by luu-alex — APPROVED** [review]
